clientv3/watch.go

```diff
@@ -94,6 +94,9 @@ type watchRequest struct {
 	key string
 	end string
 	rev int64
+	// progressNotify is for progress updates.
+	// Progress updates have zero events in WatchResponse.
+	progressNotify bool
```

Comment:
we should move the second line to the option func. Doc here is meaningless. this is a private field. user will not see this.